### PR TITLE
Update SystemDesc's ChipDesc to have Worker, Dram, Eth physical core list

### LINKED ISSUE #264: Add physical core coords to system desc
We want to use the following metal device APIs to add to system desc:
```c++
CoreCoord logical_grid_size() const;
CoreCoord worker_core_from_logical_core(const CoreCoord &logical_core) const;

CoreCoord dram_grid_size() const;
CoreCoord dram_core_from_logical_core(const CoreCoord &logical_core) const;
```

This should populate a new flatbuffer fields:
```
table CoreMapping {
    logical: Dim2d;
    physical: Dim2d;
};

table SystemDesc {
    ...
   worker_core_mapping: [CoreMapping];
   dram_core_mapping: [CoreMapping];
};
```

Blocked by #132, once 132 is complete it will be much easier to make updates to system desc, because we can remove system desc from all of the tests.

### PR DESCRIPTION
**Pull request by kmabeeTT**
Closes #264

First TTMLIR PR finally, hooray.

from commit desc:

 - Originally was going to have logical to phyical cores by type, but after some discussion, storing physical cores only, by type under ChipDesc
 - Create TT_CoreCoordAttr layer w/ Y,X members to make printing nicer and in the format [(1, 2), (2, 3)] for 2 core coordinates.
 
 
 Tested with below commands:
 
 
 ```
 
% ttrt query --save-artifacts --system-desc // Looked at system desc, looks good
%  ./build/bin/ttmlir-opt --ttir-to-ttnn-backend-pipeline --ttir-load-system-desc="path=./ttrt-artifacts/system_desc.ttsys" test/ttmlir/Dialect/TTNN/simple_subtract.mlir

#l1_ = #tt.memory_space<l1>
#system = #tt.memory_space<system>
#layout = #tt.layout<(d0, d1) -> (d0, d1), undef, <1x1>, memref<64x128xf32, #system>>
#layout1 = #tt.layout<(d0, d1) -> (d0, d1), undef, <8x8>, memref<8x16xf32, #system>>
#layout2 = #tt.layout<(d0, d1) -> (d0, d1), undef, <8x8>, memref<8x16xf32, #l1_>>
module attributes {tt.device = #tt.device<#tt.grid<8x8, (d0, d1) -> (0, d0, d1)>, [0]>, tt.system_desc = #tt.system_desc<[{arch = <wormhole_b0>, grid = 7x8, l1_size = 1499136, num_dram_channels = 12, dram_channel_size = 1073741824, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32, physical_cores = {worker = [(2, 1), (2, 2), (2, 3), (2, 4), (2, 6), (2, 7), (2, 8), (2, 9), (3, 1), (3, 2), (3, 3), (3, 4), (3, 6), (3, 7), (3, 8), (3, 9), (4, 1), (4, 2), (4, 3), (4, 4), (4, 6), (4, 7), (4, 8), (4, 9), (5, 1), (5, 2), (5, 3), (5, 4), (5, 6), (5, 7), (5, 8), (5, 9), (7, 1), (7, 2), (7, 3), (7, 4), (7, 6), (7, 7), (7, 8), (7, 9), (8, 1), (8, 2), (8, 3), (8, 4), (8, 6), (8, 7), (8, 8), (8, 9), (9, 1), (9, 2), (9, 3), (9, 4), (9, 6), (9, 7), (9, 8), (9, 9), (10, 1), (10, 2), (10, 3), (10, 4), (10, 6), (10, 7), (10, 8), (10, 9)], dram = [(11, 0), (1, 0), (5, 0), (7, 0), (1, 5), (11, 5), (2, 5), (9, 5), (8, 5), (3, 5), (5, 5), (7, 5)], eth = [(6, 9), (6, 1)], eth_inactive = [(6, 7), (6, 2), (6, 8), (0, 4), (0, 6), (0, 3), (0, 7), (0, 2), (0, 8), (6, 6), (0, 1), (6, 3), (0, 9)]}}, {arch = <wormhole_b0>, grid = 7x8, l1_size = 1499136, num_dram_channels = 12, dram_channel_size = 1073741824, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32, physical_cores = {worker = [(2, 1), (2, 2), (2, 3), (2, 4), (2, 6), (2, 7), (2, 8), (2, 9), (3, 1), (3, 2), (3, 3), (3, 4), (3, 6), (3, 7), (3, 8), (3, 9), (4, 1), (4, 2), (4, 3), (4, 4), (4, 6), (4, 7), (4, 8), (4, 9), (5, 1), (5, 2), (5, 3), (5, 4), (5, 6), (5, 7), (5, 8), (5, 9), (7, 1), (7, 2), (7, 3), (7, 4), (7, 6), (7, 7), (7, 8), (7, 9), (8, 1), (8, 2), (8, 3), (8, 4), (8, 6), (8, 7), (8, 8), (8, 9), (9, 1), (9, 2), (9, 3), (9, 4), (9, 6), (9, 7), (9, 8), (9, 9), (10, 1), (10, 2), (10, 3), (10, 4), (10, 6), (10, 7), (10, 8), (10, 9)], dram = [(11, 0), (1, 0), (5, 0), (7, 0), (1, 5), (11, 5), (2, 5), (9, 5), (8, 5), (3, 5), (5, 5), (7, 5)], eth = [(0, 9), (0, 1)], eth_inactive = [(6, 4), (0, 8), (0, 2), (0, 7), (0, 3), (0, 6), (0, 4), (6, 9), (6, 1), (6, 8), (6, 2), (6, 7), (6, 3), (6, 6)]}}], [0, 1], [<pcie|host_mmio>, <pcie>], [<0, 0, 0, 0>]>} {
```

and a test:

```
./build/bin/ttmlir-opt --ttir-to-ttnn-backend-pipeline --ttir-load-system-desc="path=./ttrt-artifacts/system_desc.ttsys" test/ttmlir/Dialect/TTNN/simple_subtract.mlir | ./build/bin/ttmlir-translate  --ttnn-to-flatbuffer -o out.ttnn
TTMLIR_SUBTRACT_FB_PATH=/localdev/kmabee/mlir2/out.ttnn ctest --test-dir ./build/runtime/test -R TTNNSubtract --verbose |& tee run_test.log
<passes>
```
 
 Wanted to run thse, but I only have N300 for now and they fai with "only single chip supported for now" assert.
 
```
export SYSTEM_DESC_PATH=/some/path
llvm-lit -v test/ttmlir/Silicon
ttrt run build/test/ttmlir/Silicon
```

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]
Kyle this looks perfect!  I appreciate the testing and validation you did too, once we have silicon CI we'll have those commands you ran running in CI.

**Comment by nsmithtt** [comment]
@kmabeeTT, you have some CI failures.  To repro CI run this command:
```
cmake --build build -- check-ttmlir
```

I think they're all hitting the same bug.  We have some straggling tests that still have a hard coded `tt.system_desc` in them.  We need to migrate the rest of these tests, the transformation is to just remove all module attributes so:
```
module attributes {tt.system_desc = #tt.system_desc<[{arch = <wormhole_b0>, grid = 8x8, l1_size = 1048576, num_dram_channels = 12, dram_channel_size = 1048576, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32}], [0], [<pcie|host_mmio>], [<0, 0, 0, 0>]>} {
```
Becomes just:
```
module {
```

And then at the top the `RUN` directive should have this pass pre-pended `--ttir-load-system-desc`:
```
// RUN: ttmlir-opt --ttir-load-system-desc
```

See `test/ttmlir/Dialect/TTIR/test_grid_set.mlir` as an example.

**Comment by kmabeeTT** [comment]
> @kmabeeTT, you have some CI failures. 

Thanks for review! Yep, spotted the same, and talked a bit offline about it (though I was worried about risk of removing the hardcoded systemdesc from the 6 test .mlir files, so I updated them locally to paste in the "default generated" physical_cores systemdesc chunk to make them pass).  Okay, sounds like that was not the right idea, I'll try your suggestion and push a fix.

File: runtime/lib/common/system_desc.cpp
Comment (nsmithtt) [inline runtime/lib/common/system_desc.cpp:104]: `logical_grid_size` -> `compute_with_storage_grid_size`

**Review by nsmithtt — CHANGES_REQUESTED** [review]

**Review by nobradovictt — APPROVED** [review]

File: runtime/lib/common/system_desc.cpp
Comment (kmabeeTT) [reply runtime/lib/common/system_desc.cpp:104]: @nsmithtt  - Thx, okay, this is different than original idea in ticket, will change. But since `logical_grid_size()` in metal returned `worker_grid_size`, I think it makes sense to rename "workers" on our side to "compute" if we go with this, so we'd have:

```
compute
dram
eth
eth_inactive
```

Thoughts?

Comment (kmabeeTT) [reply runtime/lib/common/system_desc.cpp:104]: Yeah? I was wondering if we liked auto.  The 80 character limit was such a bother, I was doing everything I could everywhere to avoid line-braks for the simplest of code, but can change this to CoreCoord, yep.

Comment (nobradovictt) [reply runtime/lib/common/system_desc.cpp:104]: I personally avoid using it, much easier later on to figure out what is code doing. I only use it for iterators and really long types which need namespace reference.

Comment (kmabeeTT) [reply runtime/lib/common/system_desc.cpp:104]: On the fence about the rename, since we still have to call this API..

```
device->worker_core_from_logical_core()
```

Comment (nsmithtt) [reply runtime/lib/common/system_desc.cpp:104]: I think it's ok to leave them as worker, I'll leave it up to you.  The issue is that `logical_grid_size` returns the full size, which fast dispatch will take a row from. `compute_with_storage_grid_size` returns the cores you can actually use.

Comment (kmabeeTT) [reply runtime/lib/common/system_desc.cpp]: got it, thx, updated to use new func, left as worker. resolved

Comment (kmabeeTT) [reply runtime/lib/common/system_desc.cpp]: sounds good, updated, resolved.

**Comment by kmabeeTT** [comment]
Updated w/ feedback, rebased on latested (saw 4 of 6 test .mlir files, transpose related got fixed this morning, so updated commit desc since I no longer modify them), confirmed this runs locally (hacked N300 to only use 1 chip):

```
export SYSTEM_DESC_PATH=/some/path
llvm-lit -v test/ttmlir/Silicon
ttrt run build/test/ttmlir/Silicon
```

 will merge after @nsmithtt re-review approval

**Review by nsmithtt — APPROVED** [review]
Thanks!

**Comment by kmabeeTT** [comment]
> Just a reminder for the 24hour rule

Crap, right. I rounded 16 hours to 24, will be more careful.

**Pull request merged**
